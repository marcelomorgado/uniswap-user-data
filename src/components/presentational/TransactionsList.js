import React from "react";
import PropTypes from "prop-types";
import Transaction from "./Transaction";
import InfinityList from "./InfinityList";

export default function TransactionsList({
  hasNextPage,
  isNextPageLoading,
  items,
  loadNextPage,
}) {
  // eslint-disable-next-line react/prop-types
  const rowRenderer = ({ item, style }) => {
    return <Transaction transaction={item} style={style} />;
  };

  return (
    <InfinityList
      items={items}
      loadNextPage={loadNextPage}
      hasNextPage={hasNextPage}
      isNextPageLoading={isNextPageLoading}
      rowRenderer={rowRenderer}
      rowHeight={30}
    />
  );
}

TransactionsList.propTypes = {
  hasNextPage: PropTypes.bool.isRequired,
  isNextPageLoading: PropTypes.bool.isRequired,
  loadNextPage: PropTypes.func.isRequired,
  items: PropTypes.array.isRequired,
};
