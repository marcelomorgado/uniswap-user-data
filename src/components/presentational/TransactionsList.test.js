import React from "react";
import { shallow } from "enzyme";
import TransactionsList from "./TransactionsList";

it("renders the component", () => {
  const props = {
    hasNextPage: false,
    isNextPageLoading: false,
    loadNextPage: () => {},
    items: [],
  };

  expect(shallow(<TransactionsList {...props} />)).toMatchSnapshot();
});
