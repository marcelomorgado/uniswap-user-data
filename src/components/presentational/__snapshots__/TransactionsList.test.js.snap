// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<InfiniteList
  hasNextPage={false}
  isNextPageLoading={false}
  items={Array []}
  loadNextPage={[Function]}
  rowHeight={30}
  rowRenderer={[Function]}
/>
`;
